base/src/main/java/io/spine/code/proto/ref/BuiltIn.java

```diff
@@ -0,0 +1,170 @@
+     * declaration:
+     * <pre>{@code
+     * message MyEnrichment {
+     *    string enrichment_value = 1 [(by) = "*.my_id"];
+     * }}
+     * </pre>
+     * tells that the enrichment field is populated after the a field named {@code my_id}
+     * found in any type. This, in turn, means that the type {@code MyEnrichment} will be used
+     * for all the messages declaring a field with such name.
+     *
+     * <p>A special care should be taken for ensuring that such fields have the same type, and
+     * that an enrichment function is provided for obtaining enrichment field value from a source
+     * message field.
+     */
+    ALL("*") {
+
+        /** Accepts all message types. */
+        @Override
+        public boolean test(Descriptor message) {
+            return true;
+        }
+    },
+
+    SELF("") {
+
+        /**
+         * Accepts all message types.
+         *
+         * <p>Since any message can reference its field, this method always returns {@code true}
+         * since validity of a field reference for such a message is outside of scope of
+         * responsibilities of {@link TypeRef}.
+         * @see FieldRef
+         */
+        @Override
+        public boolean test(Descriptor message) {
+            return true;
+        }
+    },
+
+    /**
+     * A reference to an event context.
+     */
+    EVENT_CONTEXT("context") {
+
+        @Override
+        Optional<TypeRef> parse(String value) {
+            if (value.startsWith(value())) {
+                return Optional.of(this);
+            }
+            return Optional.empty();
+        }
+
+        /**
+         * Accepts a message which type name is {@code EventContext}.
+         */
+        @Override
+        public boolean test(Descriptor message) {
+            return "EventContext".equals(message.getName());
+        }
+    };
+
+    /** The value of the reference. */
+    private final String value;
+
+    BuiltIn(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Matches the passed string to see if it represents this type reference.
+     *
+     * @return {@code this} if the string matches this type reference,
+     *         empty {@code Optional} otherwise
+     */
+    Optional<TypeRef> parse(String value) {
+        if (value().equals(value)) {
+            return Optional.of(this);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Ensures that the passed value is:
+     * <ol>
+     * <li>not null
+     * <li>not empty or blank
+     * <li>not a wild card type reference in a suffix form
+     *  (such as {@code '*CommonEventNameSuffix.field_name'}, which is not currently supported.
+     * </ol>
+     */
+    @CanIgnoreReturnValue
+    static String checkTypeReference(String typeReference) {
+        checkNotEmptyOrBlank(typeReference);
+        String wildcard = ALL.value();
+        if (typeReference.startsWith(wildcard)) {
+            checkArgument(
```

Comment:
It's a part of `FieldRef` logic. Please see tests for this class. It is possible to add such a reference, meaning that only `*` represents `ALL` type reference. Where “all” can mean types referenced in `(enrichment_for)`. If there is no such option defined, it would really mean “all types” with such a field.